From: Sophie Brun <sophie@offensive-security.com>
Date: Tue, 26 Oct 2021 10:21:00 +0200
Subject: Remove failing m4_undefine([backend])

---
 libssh2/configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libssh2/configure.ac b/libssh2/configure.ac
index 686dddb..690feed 100644
--- a/libssh2/configure.ac
+++ b/libssh2/configure.ac
@@ -127,7 +127,7 @@ fi
 m4_set_foreach([crypto_backends], [backend],
   [AM_CONDITIONAL(m4_toupper(backend), test "$found_crypto" = "backend")]
 )
-m4_undefine([backend])
+#m4_undefine([backend])
 
 
 # libz
